test/AspNetCoreModule.Test/Framework/InitializeTestMachine.cs

```diff
@@ -65,27 +83,135 @@ public class InitializeTestMachine : IDisposable
+                    isElevated = principal.IsInRole(WindowsBuiltInRole.Administrator);
+
+                    // check if this test process is started with the Run As Administrator start option
+                    _globalTestFlags = Environment.ExpandEnvironmentVariables(ANCMTestFlagsEnvironmentVariable);
+                    _globalTestFlags = _globalTestFlags.ToLower();
+                    if (_globalTestFlags.Contains(ANCMTestFlags.UseIISExpressContext.ToLower()) && _globalTestFlags.Contains(ANCMTestFlags.UseFullIISContext.ToLower()))
+                    {
+                        throw new ApplicationException(ANCMTestFlagsEnvironmentVariable + " can't be set with both "
+                            + ANCMTestFlags.UseIISExpressContext + " and " + ANCMTestFlags.UseFullIISContext);
+                    }
+
+                    // set default context with ANCMTestFlags_DefaultContext_IISExpressP0Only
+                    if (!isElevated
+                        || _globalTestFlags.Replace(ANCMTestFlagsEnvironmentVariable.ToLower(), "").Trim() == ""
+                        || _globalTestFlags.Contains(ANCMTestFlags.UseIISExpressContext.ToLower()))
+                    {
+                        _globalTestFlags = ANCMTestFlags_DefaultContext_IISExpressP0Only;
+                    }
+
+                    // if users set UseFullIIS, the default value should be overwritten with ANCMTestFlags_DefaultContext_FullIIS
+                    if (isElevated
+                        && _globalTestFlags.Contains(ANCMTestFlags.UseFullIISContext.ToLower()))
+                    {
+                        _globalTestFlags = ANCMTestFlags_DefaultContext_FullIIS;
+                    }
+
+                    // adjust the default test context in run time to figure out wrong test context values
+                    if (!isElevated)
+                    {
+                        if (_globalTestFlags.Contains(ANCMTestFlags.UseFullIISContext.ToLower()))
+                        {
+                            _globalTestFlags = _globalTestFlags.Replace(ANCMTestFlags.UseFullIISContext.ToLower(), "");
+                        }
+
+                        if (!_globalTestFlags.Contains(ANCMTestFlags.UseIISExpressContext.ToLower()))
+                        {
+                            TestUtility.LogInformation("Added test context of " + ANCMTestFlags.UseIISExpressContext);
+                            _globalTestFlags += ";" + ANCMTestFlags.UseIISExpressContext;
+                        }
+
+                        if (!_globalTestFlags.Contains(ANCMTestFlags.RunAsNonAdminUser.ToLower()))
+                        {
+                            TestUtility.LogInformation("Added test context of " + ANCMTestFlags.RunAsNonAdminUser);
+                            _globalTestFlags += ";" + ANCMTestFlags.RunAsNonAdminUser;
+                        }
+                    }
+
+                    if (MakeCertExeAvailable == true)
```

Comment:
nit: remove true check.

Reply:
It is required because this variable can be set with null, false or true.

Reply:
No like just if(MakeCertExeAvailable)

Reply:
This is already fixed. I guess you looked at the old version.